tests/core/test_domain.py

```diff
@@ -802,3 +803,12 @@ def test_domain_utterance_actions_deprecated_templates():
     old_domain = Domain.from_yaml(old_yaml)
     new_domain = Domain.from_yaml(new_yaml)
     assert hash(old_domain) == hash(new_domain)
+
+
+def test_domain_from_dict_does_not_change_input():
```

Comment:
this doesn't cover the case where the intents dictionary is changed, right? Optional: Make sure it's also covered

Reply:
I replaced loading a domain from the data folder by hardcoding a simple one which contains both slots and intents that might get changed.

Reply:
slick!